sdk/core/azure-core/src/main/java/com/azure/android/core/http/interceptors/UserAgentInterceptor.java

```diff
@@ -0,0 +1,152 @@
+    private static final String DEFAULT_USER_AGENT_HEADER = "azsdk-java";
+
+    // From the design guidelines, the default user agent header format is:
+    // azsdk-java-<client_lib>/<sdk_version> <platform_info>
+    private static final String USER_AGENT_FORMAT = DEFAULT_USER_AGENT_HEADER + "-%s/%s (%s)";
+
+    // From the design guidelines, the application id user agent header format is:
+    // AzCopy/10.0.4-Preview azsdk-java-<client_lib>/<sdk_version> <platform_info>
+    private static final String APPLICATION_ID_USER_AGENT_FORMAT = "%s " + USER_AGENT_FORMAT;
+
+    // When the AZURE_TELEMETRY_DISABLED configuration is true remove the <platform_info> portion of the user agent.
+    private static final String DISABLED_TELEMETRY_USER_AGENT_FORMAT = DEFAULT_USER_AGENT_HEADER + "-%s/%s";
+
+    // From the design guidelines, the platform info format is:
+    // <language runtime>; <os name> <os version>
+    private static final String PLATFORM_INFO_FORMAT = "%s; %s %s";
+
+    private final String userAgent;
+
+    /**
+     * Creates a {@link UserAgentInterceptor} with a default user agent string.
+     */
+    public UserAgentInterceptor() {
+        this(null);
+    }
+
+    /**
+     * Creates a UserAgentInterceptor with {@code userAgent} as the header value. If {@code userAgent} is {@code null},
+     * then the default user agent value is used.
+     *
+     * @param userAgent The user agent string to add to request headers.
+     */
+    public UserAgentInterceptor(String userAgent) {
+        if (userAgent != null) {
+            this.userAgent = userAgent;
+        } else {
+            this.userAgent = DEFAULT_USER_AGENT_HEADER;
+        }
+    }
+
+    /**
+     * Creates a UserAgentInterceptor with the {@code sdkName} and {@code sdkVersion} in the User-Agent header value.
+     *
+     * If the passed configuration contains true for AZURE_TELEMETRY_DISABLED the platform information won't be included
+     * in the user agent.
+     *
+     * @param applicationId User specified application Id.
+     * @param sdkName Name of the client library.
+     * @param sdkVersion Version of the client library.
+     * @param version {@link ServiceVersion} of the service to be used when making requests.
+     * @param configuration Configuration store that will be checked for the AZURE_TELEMETRY_DISABLED.
+     * @throws NullPointerException if {@code configuration} is {@code null}.
+     */
+    public UserAgentInterceptor(String applicationId, String sdkName, String sdkVersion, Configuration configuration,
+                           ServiceVersion version) {
+        Objects.requireNonNull(configuration, "'configuration' cannot be null.");
+        boolean telemetryDisabled = configuration.get(Configuration.PROPERTY_AZURE_TELEMETRY_DISABLED, false);
+        if (telemetryDisabled) {
+            this.userAgent = String.format(DISABLED_TELEMETRY_USER_AGENT_FORMAT, sdkName, sdkVersion,
```

Comment:
Is there any reason for using one extra argument (`version.getVersion()`) at the end of the formatting call? All statements where we set the `userAgent` through the use `String.format()` have this additional element as a parameter.

Reply:
Will be removing these for now. A deeper discussion about how to handle telemetry on mobile needs to be had though.